File: 167. Two Sum II - Input Array Is Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        dic = {}
        for index,num in enumerate(numbers):
            # 如果那個值在裡面，那就回傳
            if (target-num in dic):
                return[dic[target-num],index+1]
            # 如果沒有就加進map
            dic[num]=index+1

File: C++/自己搞定/Medium/167. Two Sum II - Input Array Is Sorted.cpp
#include<iostream>
#include <bits/stdc++.h>


class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& numbers, int target) {
        int left = 0;int right = numbers.size()-1;std::vector<int> answer;
        if (numbers.size()==2){
            answer = {1,2};
            return answer;
        }
        while(right>left){
            if (numbers[right]+numbers[left]>target){
                right--;
            }

            else if (numbers[right]+numbers[left]<target){
                left++;
            }
            else{
                answer.push_back(left+1);
                answer.push_back(right+1);
                break;
            }

        }
        return answer;
    }
};
